modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -632,6 +632,23 @@ FeatureFloodCount::doesFeatureIntersectBoundary(unsigned int feature_id) const
   return false;
 }
 
+Point
+FeatureFloodCount::featureCentroid(unsigned int feature_id) const
+{
+  Point p;
+  if (feature_id >= _feature_id_to_local_index.size())
+    return p;
```

Comment:
I'm not sure I'd return a (0,0,0) where processors may be missing information. That point usually is in the domain so it's hard to tell a valid result from an invalid one. As you can see with similar getters, usually the information isn't complete so I have a lot of protection built in to prevent people from getting incorrect information when coupling. If you need me to help out with making sure that you can obtain Centroids in all cases, let me know. I'm not entirely sure if this code is completely safe in parallel or not without a little looking around.